### PR TITLE
M3-2080 Login As Customer

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Login as Customer

## Type of Change
- Non breaking change ('update', 'change')

## Applicable E2E Tests

N/A

## Note to Reviewers

You must have a functioning devenv to get this working. Please reach out if you have trouble getting the devenv up and running.

Once you have devenv up and running, reach out for an INSERT statement

Once you have devenv and manager running, navigate to `http://localhost:3000/admin/callback#access_token=ec592ecb23e08bbcc0c8008b0898858436df564e04e8975603fdfbe1238e4406&destination=linodes&token_type=Admin&expires_in=7200`

### CODE REVIEW

**Comment by martinmckenna** [comment]
Screen Shot 2019-04-09 at 3 40 53 PM

File: src/store/authentication/authentication.reducer.ts [hunk 1]
```diff
@@ -40,6 +43,24 @@ const reducer = reducerWithInitialState(defaultState)
       expiration: expires || null
     };
   })
+  .case(handleStartSessionAsCustomer, (state, payload) => {
+    const { token, expires } = payload;
+
+    /** set local storage */
```
Comment (johnwcallahan) [inline src/store/authentication/authentication.reducer.ts]: Do we have to set local storage here? I think it's fine (and even preferred) if sessions don't persist across refreshes/new tabs, etc.

Comment (Jskobos) [reply src/store/authentication/authentication.reducer.ts]: I agree, I think Will Smith was pretty adamant about never persisting any of this.

Comment (martinmckenna) [reply src/store/authentication/authentication.reducer.ts]: spoke to Will here and we're going to store this in local storage after all. I updated some of the logic to not include an extra local storage variable.

So now the `token` key in local storage will look like `Admin 123131` or `Bearer 3214235`

File: src/layouts/LoginAsCustomerCallback.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+type CombinedProps = DispatchProps & RouteComponentProps;
+
+interface QueryParams {
+  access_token: string;
+  token_type: string;
+  destination: string;
+  expires_in: string;
+}
+
+export class OAuthCallbackPage extends PureComponent<CombinedProps> {
```
Comment (johnwcallahan) [inline src/layouts/LoginAsCustomerCallback.tsx]: Should probably change the name of this component.

Also, I think it'd be helpful to add a comment explaining that this component is 90% similar to the one in `OAuth.tsx` (which I think is fine, in this case).

File: src/store/authentication/authentication.reducer.ts [hunk 3]
```diff
@@ -45,8 +46,10 @@ const reducer = reducerWithInitialState(defaultState)
      * if our token is expired, clear local storage
      * and redux state
      */
-    const expiryTime = expiryInLocalStorage.get();
-    if (expiryTime && new Date(expiryTime) < new Date()) {
+    const expiryTimeFromLocalStorage = expiryInLocalStorage.get();
+    const expiryDate = new Date();
+    expiryDate.setTime(expiryDate.getTime() + +expiryTimeFromLocalStorage);
```
Comment (johnwcallahan) [inline src/store/authentication/authentication.reducer.ts]: `expiryDate` is coming up as `Invalid Date` here.

**Pull request merged**